resources/projectsyncup-design-document.md

```diff
@@ -0,0 +1,199 @@
+
+## 6. API
+
+### 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ProjectModel
+String projectId;
+String projectName;
+String projectStatus;
+List<Task> projectTasks;
+List<UserModel> projectTeamMembers;
+```
+
+```
+// TaskModel
+String taskId;
+String taskName;
+String taskDescription;
+String taskDueDate;
+String taskStatus;
+List<UserModel> taskAssignedUsers;
+Integer taskTimer;
+```
+
+```
+// UserModel
+String userId;
+String userName;
+String userRole;
+```
+
+
+### 6.3. _Create Project Endpoint_
+* Accepts `POST` requests to `/projects`
+* Accepts data to create a new Project with a given name, project status, and an optional list of tasks. Returns the new project, including a unique project ID assigned by the Project Management Service.
+* Request Body: <br>
+    ```"projectName": "Project Name", ```<br>
+    ```"projectStatus": "Not Started", ```<br>
+    ```"projectStatus": "Not Started", ```<br>
+    ```"projectTeamMembers": ["userId_1", "userId_2"] ```<br>
+
+### 6.4. _Create Task Endpoint_
+* Accepts `POST` requests to `/tasks`
+* Accepts data to create a new Task with a given task name, description, due date, task status, and list of assigned team members. Returns the new Task, including a unique task ID assigned by the Project Management Service.
+* Request Body: <br>
+    ```{"taskName": "Task Title",``` <br>
+    ```"taskDescription": "Task Description", ```<br>
+    ```"taskStatus": "Not Started", ```<br>
+    ```"taskAssignedUsers": ["userId_1", "userId_2"], ```<br>
+    ```"taskStatus": "open", ```<br>
+    ```"taskDueDate": "yyyy-mm-dd", ```<br>
+    ```"taskTimer": 60```<br>
+
+![](images/create-task-activity-SD.png)
+
+### 6.5. _Add Task To Project Endpoint_
+* Accepts `PUT` requests to `/projects/:id/task`
+* Accepts a project ID and task ID and adds the task to the specified project. 
+  * If the project is not found, will throw a ProjectNotFoundException 
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+### 6.6. _Update Task Information Endpoint_
+* Accepts `PUT` requests to `/tasks/:id`
+* Accepts a task ID and updated task information, Returns the updated the task with the new information.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+* Request Body: <br>
+  ```{"taskName": "Task Title: New Title",``` <br>
+  ```"taskDescription": " Updated Task Description", ```<br>
+  ```"taskAssignedUsers": ["userId_1", "userId_2"], ```<br>
+  ```"taskStatus": "Not Started", ```<br>
+  ```"taskDueDate": "yyyy-mm-dd", ```<br>
+  ```"taskTimer": 60```<br>
+
+### 6.7. _Update Task Progress Endpoint_
```

Comment:
I don't think you need this one or 6.8 - you can combine both of them into 6.6. That endpoint should update the task based on whatever fields are submitted to it (whether the status or time spent, or other details).